Meadowlark-js/packages/meadowlark-core/src/validation/DocumentIdentityExtractor.ts

```diff
@@ -62,7 +72,12 @@ function documentIdentitiesFrom(
   if (isReferenceElement(identityReferenceComponent)) {
     // SchoolYearEnumerations are an API one-off expressed as two levels in the document body
     if (identityReferenceComponent.sourceProperty.type === 'schoolYearEnumeration') {
-      documentPath.push('schoolYearTypeReference');
+      const { roleName } = identityReferenceComponent.sourceProperty;
+      if (roleName != null && roleName.trim().length > 0) {
```

Comment:
As above, not wrong to do a roleName check this way but the type of EntityProperty.roleName (like almost everything in the MetaEd model) is not nullable. Since the string is not nullable, the absence of a string value in the MetaEd model is the empty string, which simplifies these kind of checks.